dev/com.ibm.ws.microprofile.config_fat/fat/src/com/ibm/ws/microprofile/config/fat/tests/ClassLoaderCacheTest.java

```diff
@@ -0,0 +1,134 @@
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import com.ibm.websphere.simplicity.ShrinkHelper;
+import com.ibm.ws.microprofile.appConfig.classLoaderCache.test.ClassLoaderCacheTestServlet;
+import com.ibm.ws.microprofile.config.fat.suite.SharedShrinkWrapApps;
+
+import componenttest.annotation.Server;
+import componenttest.custom.junit.runner.FATRunner;
+import componenttest.topology.impl.LibertyServer;
+import componenttest.topology.utils.FATServletClient;
+
+/**
+ *
+ */
+@RunWith(FATRunner.class)
+public class ClassLoaderCacheTest extends FATServletClient {
+
+    public static final String APP_NAME = "classLoaderCache";
+
+    public static final String EARA = APP_NAME + "A";
+    public static final String EARB = APP_NAME + "B";
+    public static final String EARA_NAME = EARA + ".ear";
+    public static final String EARB_NAME = EARB + ".ear";
+
+    public static final String WARA1 = EARA + "1";
+    public static final String WARA2 = EARA + "2";
+    public static final String WARA1_NAME = WARA1 + ".war";
+    public static final String WARA2_NAME = WARA2 + ".war";
+
+    public static final String WARB1 = EARB + "1";
+    public static final String WARB2 = EARB + "2";
+    public static final String WARB1_NAME = WARB1 + ".war";
+    public static final String WARB2_NAME = WARB2 + ".war";
+
+    @Server("ClassLoaderCacheServer")
+    public static LibertyServer server;
+
+    @BeforeClass
+    public static void setUp() throws Exception {
+        JavaArchive testAppUtils = SharedShrinkWrapApps.getTestAppUtilsJar();
+
+        WebArchive classLoaderCacheA1_war = ShrinkWrap.create(WebArchive.class, WARA1_NAME)
+                        .addAsLibrary(testAppUtils)
+                        .addPackage("com.ibm.ws.microprofile.appConfig.classLoaderCache.test");
+
+        WebArchive classLoaderCacheA2_war = ShrinkWrap.create(WebArchive.class, WARA2_NAME)
+                        .addAsLibrary(testAppUtils)
+                        .addPackage("com.ibm.ws.microprofile.appConfig.classLoaderCache.test");
+
+        WebArchive classLoaderCacheB1_war = ShrinkWrap.create(WebArchive.class, WARB1_NAME)
+                        .addAsLibrary(testAppUtils)
+                        .addPackage("com.ibm.ws.microprofile.appConfig.classLoaderCache.test");
+
+        WebArchive classLoaderCacheB2_war = ShrinkWrap.create(WebArchive.class, WARB2_NAME)
+                        .addAsLibrary(testAppUtils)
+                        .addPackage("com.ibm.ws.microprofile.appConfig.classLoaderCache.test");
+
+        EnterpriseArchive classLoaderCacheA_ear = ShrinkWrap.create(EnterpriseArchive.class, EARA_NAME)
+                        .addAsManifestResource(new File("test-applications/" + EARA_NAME + "/resources/META-INF/application.xml"), "application.xml")
+                        .addAsManifestResource(new File("test-applications/" + EARA_NAME + "/resources/META-INF/permissions.xml"), "permissions.xml")
```

Comment:
These could probably be kept as resources in the same package, rather than loaded as files from test-applications if you preferred.